docs/aws/audit/redshiftmonitoring/rules/unused_redshift_reserved_nodes.mdx

```diff
@@ -24,6 +24,149 @@ CBP
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to install and configure AWS CLI on your local machine. You can download it from the official AWS website and configure it using the "aws configure" command. You will be prompted to provide your AWS Access Key ID, Secret Access Key, Default region name, and Default output format.
+
+2. Once the AWS CLI is set up, you can list all the Redshift Reserved Nodes using the following command:
+
+   ```
+   aws redshift describe-reserved-nodes
+   ```
+
+   This command will return a JSON output with details of all the reserved nodes.
+
+3. To check if a Reserved Node is unused, you need to check the "State" attribute in the output. If the "State" attribute is "available", it means that the Reserved Node is not currently being used.
+
+4. To automate this process, you can use a Python script with the Boto3 library (AWS SDK for Python) to list all the Reserved Nodes and check their states. Here is a simple script:
+
+   ```python
+   import boto3
+
+   # Create a Redshift client
+   redshift = boto3.client('redshift')
+
+   # List all reserved nodes
+   reserved_nodes = redshift.describe_reserved_nodes()
+
+   # Check if any reserved node is unused
+   for node in reserved_nodes['ReservedNodes']:
+       if node['State'] == 'available':
+           print(f"Unused Reserved Node: {node['ReservedNodeId']}")
+   ```
+
+   This script will print the IDs of all unused Reserved Nodes.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install and configure AWS SDK for Python (Boto3) on your local system. Boto3 allows you to directly create, update, and delete AWS services from your Python scripts.
+
+2. Import the necessary libraries and establish a session using your AWS credentials. Here's a sample script to get started:
+
+```python
+import boto3
+
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
+    region_name='YOUR_REGION'
+)
+```
+
+3. Now, create a Redshift client using the session created above. Then, use the `describe_reserved_nodes` method to get a list of all reserved nodes.
+
+```python
+redshift_client = session.client('redshift')
+
+response = redshift_client.describe_reserved_nodes()
+
+reserved_nodes = response['ReservedNodes']
+```
+
+4. Iterate over the list of reserved nodes and check the `State` of each node. If the `State` is `unused`, then the reserved node is not being used.
+
+```python
+for node in reserved_nodes:
+    if node['State'] == 'unused':
+        print(f"Unused Reserved Node: {node['ReservedNodeId']}")
+```
+
+This script will print the IDs of all unused reserved nodes. You can modify this script according to your needs, for example, to send an alert or report if any unused reserved nodes are found.
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>